noir_stdlib/src/cmp.nr

```diff
@@ -8,12 +8,10 @@ impl Eq for Field { fn eq(self, other: Field) -> bool { self == other } }
 
 impl Eq for u1 { fn eq(self, other: u1) -> bool { self == other } }
 impl Eq for u8 { fn eq(self, other: u8) -> bool { self == other } }
-impl Eq for u16 { fn eq(self, other: u16) -> bool { self == other } }
```

Comment:
u16 being restricted is due to the avm not supporting u16 (not very popular), we can add this case back